examples/tic-tac-toe/ui/src/hooks/useTurnCapQuery.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useCurrentAccount, useSuiClient, useSuiClientContext } from '@mysten/dapp-kit';
+import { useQuery, useQueryClient, UseQueryResult } from '@tanstack/react-query';
+import { useNetworkVariable } from 'config';
+
+export type TurnCap = {
+	id: { id: string };
+	game: string;
+};
+
+export type UseTurnCapResult = UseQueryResult<TurnCap | null, Error>;
+export type InvalidateTurnCapQuery = () => void;
+
+/** Refetch TurnCaps every 5 seconds */
+const REFETCH_INTERVAL = 5000;
+
+/**
+ * Hook to fetch the `TurnCap` owned by `address` associated with
+ * `game`, if there is one.
+ */
+export function useTurnCapQuery(game?: string): [UseTurnCapResult, InvalidateTurnCapQuery] {
+	const suiClient = useSuiClient();
+	const queryClient = useQueryClient();
+	const ctx = useSuiClientContext();
+	const packageId = useNetworkVariable('packageId');
+	const account = useCurrentAccount();
+
+	const queryKey = [ctx.network, 'turn-cap', account?.address, game];
+	const response = useQuery({
+		enabled: !!game,
+		refetchInterval: REFETCH_INTERVAL,
+		queryKey,
+		queryFn: async () => {
+			const owner = account?.address;
+			if (!owner) {
+				return null;
+			}
+
+			for (;;) {
```

Comment:
never seen this used instead of just doing a while, had no idea this would work, `undefined`, With the condition being empty, I would have expected this to never loop..  TIL

Reply:
I believe this is a C idiom and it exists for the sole reason that it's the shortest string of characters to achieve this result:

```
while (true) {
while (1) {
for (;;) {
```

But I'm not attached to it -- if it looks weird, I'll switch it to whatever's the convention!